src/ScoreTrackerSecondary.java

```diff
@@ -0,0 +1,93 @@
+import components.map.Map;
+
+public abstract class ScoreTrackerSecondary implements ScoreTracker {
+    /**
+     * Returns the highest score among all players.
+     *
+     * @return the highest score
+     * @requires getTotalPlayers() > 0
+     * @ensures [returns the maximum score among all players in this]
+     */
+    @Override
+    public int getHighestScore() {
+        assert (playerScores.size() > 0);
```

Comment:
Where is playerScores coming from? You only have access to the kernel methods, not the representation.